docs/proposals/algorithms/multimodal-large-model-joint-learning /proposal.md

```diff
@@ -0,0 +1,410 @@
+               raise ValueError(f"Incremental rounds (value={self.incremental_rounds}) "
+                                "must be an integer and not less than 2.")
+
+       def _parse_config(self, config):
+           # Parse the configuration dictionary for TestEnv settings
+           config_dict = config[str.lower(TestEnv.__name__)]
+           for k, v in config_dict.items():
+               if k == str.lower(Dataset.__name__):
+                   self.datasets.append(Dataset(v))  # Initialize datasets
+               elif k == "modalities":
+                   self.modalities = v  # Set modalities
+               else:
+                   if k in self.__dict__:
+                       self.__dict__[k] = v  # Update attributes dynamically
+
+           self._check_fields()  # Validate the parsed configuration
+
+       def prepare(self):
+           # Prepare datasets for testing
+           try:
+               for dataset in self.datasets:
+                   dataset.process_datasets()  # Process each dataset
+           except Exception as err:
+               raise RuntimeError(f"Dataset preparation failed, error: {err}.") from err
+   ```
+
+Great! Here’s a comprehensive summary that includes the new benchmarking job configuration, algorithm configuration, and the updated metrics functions for evaluating multimodal models.
+
+---
+
+### New Benchmarking Configuration
+
+A new benchmarking job configuration file specific to MLLM has been created.
+
+```yaml
+benchmarkingjob:
+  name: "mllm_benchmarking_job"  # Name of the benchmarking job
+  workspace: "/ianvs/multimodal_language_model_bench/workspace"  # Path to the workspace
+  testenv: "./examples/mllm_benchmark/testenv.yaml"  # Path to the test environment configuration
+  test_object:
+    type: "algorithms"  # Specifies that the test object is algorithms
+    algorithms:
+      - name: "mllm_evaluation"  # Name of the MLLM evaluation algorithm
+        url: "./examples/mllm_benchmark/algorithms/mllm_algorithm.yaml"  # Path to the algorithm configuration
+  rank:
+    sort_by: 
+      - { "accuracy": "descend" }  # Primary sorting criterion by accuracy
+      - { "f1_score": "descend" }   # Secondary sorting criterion by F1 score
+      - { "bleu": "descend" }        # Tertiary sorting criterion by BLEU score
+      - { "fid": "ascend" }          # Quaternary sorting criterion by FID (lower is better)
+      - { "wer": "ascend" }          # Quinary sorting criterion by WER (lower is better)
+      - { "vqa_accuracy": "descend" }  # Senary sorting criterion by VQA accuracy
+    visualization:
+      mode: "selected_only"  # Visualization mode, showing only selected items
+      method: "print_table"   # Method to visualize results
+    selected_dataitem:
+      paradigms: [ "all" ]  # Selects all paradigms
+      modules: [ "all" ]     # Selects all modules
+      hyperparameters: [ "all" ]  # Selects all hyperparameters
+      metrics: 
```

Comment:
As mentioned above, metrics should be implemented in examples to avoid impacts on others examples. 

The usage of metrics is also in [ianvs examples with testenv.yaml](https://ianvs.readthedocs.io/en/latest/user_interface/how-to-config-testenv.html#id4).  An example is available in ianvs documents, as the following.

```
# testenv.yaml
testenv:
...

# metric used for model evaluation
model_metric:
  # metric name; string type;
  name: "f1_score"
  # the url address of python file
  url: "./examples/pcb-aoi/incremental_learning_bench/testenv/f1_score.py"
```